OrrRock/OrrRock/RouteFindingUpload/RouteFindingCameraViewController+AVCapturePhotoCaptureDelegate.swift

```diff
@@ -28,7 +28,7 @@ extension RouteFindingCameraViewController: AVCapturePhotoCaptureDelegate {
     
     func photoOutput(_ output: AVCapturePhotoOutput, didFinishCaptureFor resolvedSettings: AVCaptureResolvedPhotoSettings, error: Error?) {
         
-        guard error != nil else {
+        guard error == nil else {
```

Comment:
단순히 nil 인지를 확인하는 것이라면 guard 와 if let 의 목적성에 맞게
```suggestion
        if let error {
```
와 같이 변경하는것이 좋지 않을까요?

Reply:
`guard`와 `if` 구문 중에서 어떤 것을 사용하면 좋을지에 대해 공부한 후 다음에 추가적으로 픽스 사항 전달드리겠습니다! 리뷰 감사합니다 도도!

Reply:
제가 작성한 글을 끌올하면 아래와 같습니다! 

[스택오버플로우 글](https://stackoverflow.com/questions/32256834/swift-guard-let-vs-if-let)와 [애플 개발자 포럼 글](https://developer.apple.com/forums/thread/100137)를 보면서 guard의 사용 용례와 의도를 살펴봤을 때는 Swift 주류에 다음의 관행이 있다고 생각됩니다.

guard 문은 조건 검사에 따라 함수를 종료하거나 반복문 제어가 필수적일 때 사용한다 - if 문은 선택적으로 사용가능하다.
guard 문은 조건 검사 후 해당 조건이 함수 내 guard문 아래 코드에서 유효할 때 사용한다. if 문은 조건 검사 후 해당 조건이 if 블럭 내에서 유효하게 작용하는 코드에서 사용한다.
따라서, guard 문을 통해 함수 내 실질적인 로직을 시작하기 전에 값의 유효성을 검사하는 역할을 수행할 수 있다.
guard 문에 함수가 정상적으로 동작하기 위해 필요한 assertion들을 명시해둘 수 있다는 점에서 코드 가독성을 올릴 수 있다.
둘 다 내부적으로는 조건에 따라 분기하는 코드라는 점에서 서로 동일한 기능을 하도록 변환이 가능한 건 맞지만, 두 구문의 의미를 생각해서 적절하게 적용해야한다고 생각해요.

저는 함수의 정상적인 동작을 위해 사전에 assertion을 작성하고 값을 검사하는 역할을 guard문이 수행하며, if와 다르게 false 일 때 guard 블럭이 실행된다는 것 2가지를 통해 guard를 활용한 코드의 의미 전달력과 코드 가독성이 크게 올라간다고 생각해요. 그런데 코드를 깔끔하게 작성하기 위해 guard가 어울리지 않는 부분에 guard 문을 사용한다면, 전체적인 코드 내 guard의 의미가 모호해져서 코드 가독성과 신뢰성을 떨어뜨릴 수도 있겠다는 걱정이 드는 것 같아요.

이런 이유에서 저는 guard는 guard가 어울리는 상황에서만 써야한다고 생각합니다!

Reply:
30일까지 `guard` 와 `if` 구문 중 좀 더 적절한 표현이 무엇일지에 대해 고민하고 따로 문서를 작성하겠습니다! 코멘트 감사합니다 레드 :)

Reply:
기존의 구문 (guard error != nil else) 처럼 사용한다면 'nil이 아님을 판단하는 통상적인 구문'에 적합하다고 생각되지만,
바뀐 구문 (guard error == nil else) 처럼 사용한다면 통상적인 구문에 반대되어 보는이로 하여금 오히려 혼란을 줄수 있다고 생각되어
if let을 사용하여 통상적인 구문과 다름을 표현하는게 좋지 않을까 싶었어요.
(물론 != nil 이 저의 기준으로 통상적인것 같긴 합니다!)

레드가 말씀해주신것처럼 guard의 말그대로의 guard적인 부분에 중점을 둔다면 guard가 어울리는것도 이해가 갑니다!